document-store/src/integrationTest/java/org/hypertrace/core/documentstore/mongo/MongoQueryExecutorIntegrationTest.java

```diff
@@ -127,6 +127,32 @@ public void testFindSimple() throws IOException {
     assertDocsEqual(resultDocs, "mongo/simple_filter_response.json");
   }
 
+  @Test
+  public void testFindWithDuplicateSelections() throws IOException {
+    List<SelectionSpec> selectionSpecs =
+        List.of(
+            SelectionSpec.of(IdentifierExpression.of("item")),
+            SelectionSpec.of(IdentifierExpression.of("item")),
+            SelectionSpec.of(IdentifierExpression.of("price")),
+            SelectionSpec.of(IdentifierExpression.of("quantity")),
+            SelectionSpec.of(IdentifierExpression.of("quantity")),
+            SelectionSpec.of(IdentifierExpression.of("date")));
+    Selection selection = Selection.builder().selectionSpecs(selectionSpecs).build();
+    Filter filter =
+        Filter.builder()
+            .expression(
+                RelationalExpression.of(
+                    IdentifierExpression.of("item"),
+                    NOT_IN,
+                    ConstantExpression.ofStrings(List.of("Soap", "Bottle"))))
+            .build();
+
+    Query query = Query.builder().setSelection(selection).setFilter(filter).build();
+
+    Iterator<Document> resultDocs = collection.find(query);
+    assertDocsEqual(resultDocs, "mongo/simple_filter_response.json");
```

Comment:
Shouldn't we be returning results for duplicate selections back, instead of de-duping it in the response? Basically, if a client asks for N number of selections, it expects N number of responses, corresponding to each selection.

Cc @avinashkolluru @aaron-steinfeld

Reply:
Agree with you on this point. But, unfortunately, since Mongo returns results in the JSON format, I'm not sure how do we have duplicate keys in the same object.

A simple workaround for this is that the clients make sure to have different "aliases" for the same selection field, so that they are repeated in the response.

Reply:
And not just Mongo, the method org.hypertrace.core.documentstore.Collection.aggregate() itself returns an iterator of org.hypertrace.core.documentstore.Document objects, which exposes only one method "toJson()".

Reply:
In a traditional DB with columns/rows I agree with what you say @skjindal93 
But this is a document store interface. The response for the query is a Document. What does duplicate selections mean in a document(which is a JSON format). 
If you actually run a mongo query with duplicate projections the output contains it only once. In fact, I dont even think duplicates are possible in JSON.

Reply:
Ahh... @suresh-prakash beat me to this. But, my thoughts are also aligned with what Suresh says.

Reply:
Correct, dupe keys are not possible in JSON. We should error on dupe result keys (i.e. `alias.orElse(colName)`), support multiple identical selections with different result keys. Ran into this kind of thing a lot with attribute expression changes, we're _really_ bad about actually respecting the aliases provided.